### ISSUE TITLE
Copy-pasting links whose labels start with "th" or "td" in some HTML editor (itch.io's old Redaktor) generates garbage span instead

### ISSUE THREAD

**Issue by hsandt** [typora/typora-issues#5671]
- [X] Searched existing issues to avoid creating duplicates.

Yes, there is a closed issue that is similar: https://github.com/typora/typora-issues/issues/906 but I will reopen one to gain visibility back, and as I'm not sure if the cause if the same.

- [X] Confirmed that it can be reproduced in built-in themes without customized css.
- [X] Searched http://support.typora.io/

Yes, and https://support.typora.io/Copy-and-Paste/ was pretty useful. In fact, Copy as HTML Code is the function I need.
I'm still reporting this bug though as it'd be nice if standard copy-paste also worked out of the box.

----

**Describe the bug**
In some HTML editors like the one used by itch.io for game descriptions, etc. (apparently an old version of Redaktor), copy-pasting content containing links whose label starts with "th" or "td" fails to close tags properly (possibly due to th/td being incorrectly recognized as table tags), using symbols `&gt` for `>` and generating visible garbage `>span class="md-plain">`.

Unfortunately, I could only reproduce the bug with Typora + itch.io's own old version of Redaktor. Redaktor has been superseded by Redaktor X and Article, and even the latest version of Redaktor which you can live test [here](https://imperavi.com/redactor/examples/initialization/base-example/) doesn't have the bug.

Reversely, I tested another markdown editor in browser, StackEdit, which you can test [here](https://stackedit.io/app), and it didn't trigger the bug.

So it's basically a crossed bug between two specific apps, one of them outdated. In case you can still find the cause on your side though, I'm opening this ticket.

In parallel, I'll ask itch.io's web team to upgrade their HTML editor, which, let's be honest, has many other bugs. So don't feel too responsible for it, but if you find a fix quickly, then it's great!

**To Reproduce**

0. Create an itch.io account (sorry, there may be a way to grab an old version of Redaktor but I don't know exactly which version they are using)

1. Copy the following markdown:

```md
Links: [th](th-link) [td](td-link) [tr](tr-link) [th_more_stuff](th-link2) [td_more_stuff](td-link2)

Plain: th td tr
```

3. Paste it in Typora.
4. Copy it back (normal copy). Paste it in [itch.io](https://itch.io/) HTML editor.
5. Observe the resulting text adding weird `span class="md-plain"`

Screenshot:
![image](https://user-images.githubusercontent.com/1067772/226035339-b3dd487b-6fdb-45f6-b24a-d1e9ebd713f3.png)

In Markdown equivalent:

```
Links: >span class="md-plain"> >span class="md-plain"> [tr](https://itch.io/game/edit/1960184) >span class="md-plain"> [td_more_stuff](https://itch.io/game/edit/1960184)

Plain: th td tr
```

Or in HTML:

```html
<p><span class="md-plain md-expand">Links: </span><span class="md-meta-i-c  md-link"><a href="https://itch.io/game/edit/1960184"><span class="md-plain" <th<="" span=""></span></a></span>&gt;span class="md-plain"&gt; <span class="md-meta-i-c  md-link"><a href="https://itch.io/game/edit/1960184"><span class="md-plain" <td<="" span=""></span></a></span>&gt;span class="md-plain"&gt; <span class="md-meta-i-c  md-link"><a href="https://itch.io/game/edit/1960184"><span class="md-plain">tr</span></a></span><span class="md-plain"> </span><span class="md-meta-i-c  md-link"><a href="https://itch.io/game/edit/1960184"><span class="md-plain" <th_more_stuff<="" span=""></span></a></span>&gt;span class="md-plain"&gt; <span class="md-meta-i-c  md-link"><a href="https://itch.io/game/edit/1960184"><span class="md-plain">td_more_stuff</span></a></span>
</p>
<p><span class="md-plain">Plain: th td tr</span>
</p>
<p><span class="md-plain"></span></p>
<p></p>
```

**Expected behavior**
The clean text would be (example when pasting from another markdown editor):

Screenshot:
![image](https://user-images.githubusercontent.com/1067772/226027816-3e9974a8-c00e-4db4-9034-9f68770f1faf.png)

Markdown equivalent:
```
Links: [th](th-link) [td](td-link) [tr](tr-link) [th_more_stuff](th-link2) [td_more_stuff](td-link2)

Plain: th td tr
```

HTML:
```html
<p>Links:&nbsp;<a href="https://stackedit.io/th-link">th</a>&nbsp;<a href="https://stackedit.io/td-link">td</a>&nbsp;<a href="https://stackedit.io/tr-link">tr</a>&nbsp;<a href="https://stackedit.io/th-link2">th_more_stuff</a>&nbsp;<a href="https://stackedit.io/td-link2">td_more_stuff</a>
</p>
<p>Plain: th td tr
</p>
```

**Screenshots / Screencasts**
Posted above

**Sample Markdown File**
[th and td wrongly copied to itch HTML editor with span.md](https://github.com/typora/typora-issues/files/11006075/th.and.td.wrongly.copied.to.itch.HTML.editor.with.span.md)

**Desktop (please complete the following information):**
 - OS: Linux Ubuntu 22.04

**Typora Version**
1.5.8

**Additional context**
I'm pre-writing my game description in Typora then copying it to itch.io so I always keep a local copy of it.
I found the bug by accident because of one my link label started with "The" and thus "th".

**Workaround**
Edit > Copy as HTML Code and paste in the HTML editor in HTML mode.

**Comment by hsandt** [comment]
For information, here is the bug I reported on itch.io's side: https://itch.io/t/2726604/html-editor-bugs-and-update-redaktor-x-article-etc

**Comment by nk9** [comment]
When I paste the Markdown provided into Typora, re-copy and then paste into the Redactor [sample editor](https://imperavi.com/redactor/examples/initialization/base-example/#Demo), I get the expected links:

<img width="899" alt="image" src="https://github.com/typora/typora-issues/assets/3646730/97c94c97-d469-42ca-b545-23674eb1fffb">

Has this bug been fixed? Note that on macOS, you can find out what has been copied to the pasteboard using [Pasteboard Viewer](https://apps.apple.com/us/app/pasteboard-viewer/id1499215709?mt=12). When I copy the code from Typora in normal mode (not Source Code mode), the HTML that is copied looks like this (although this has been beautified for ease of reading):

```html
<!DOCTYPE html>
<span
    md-inline="plain"
    class="md-plain md-expand"
    style="
        box-sizing: border-box;
        text-indent: 0px;
        caret-color: rgb(51, 51, 51);
        color: rgb(51, 51, 51);
        font-family: 'Open Sans', 'Clear Sans', 'Helvetica Neue', Helvetica, Arial, 'Segoe UI Emoji', sans-serif;
        font-size: 16px;
        font-style: normal;
        font-variant-caps: normal;
        font-weight: 400;
        letter-spacing: normal;
        orphans: 4;
        text-align: start;
        text-transform: none;
        white-space: pre-wrap;
        widows: auto;
        word-spacing: 0px;
        -webkit-text-size-adjust: auto;
        -webkit-text-stroke-width: 0px;
        text-decoration: none;
    "
>
    Links:
</span>
<span
    md-inline="link"
    class="md-meta-i-c md-link"
    style="
        box-sizing: border-box;
        text-indent: 0px;
        caret-color: rgb(51, 51, 51);
        color: rgb(51, 51, 51);
        font-family: 'Open Sans', 'Clear Sans', 'Helvetica Neue', Helvetica, Arial, 'Segoe UI Emoji', sans-serif;
        font-size: 16px;
        font-style: normal;
        font-variant-caps: normal;
        font-weight: 400;
        letter-spacing: normal;
        orphans: 4;
        text-align: start;
        text-transform: none;
        white-space: pre-wrap;
        widows: auto;
        word-spacing: 0px;
        -webkit-text-size-adjust: auto;
        -webkit-text-stroke-width: 0px;
        text-decoration: none;
    "
>
    <a href="th-link" style="box-sizing: border-box; cursor: pointer; color: rgb(65, 131, 196); -webkit-user-drag: none; text-indent: 0px;">
        <span md-inline="plain" class="md-plain" style="box-sizing: border-box; text-indent: 0px;">th</span>
    </a>
</span>
<span
    md-inline="plain"
    class="md-plain"
    style="
        box-sizing: border-box;
        text-indent: 0px;
        caret-color: rgb(51, 51, 51);
        color: rgb(51, 51, 51);
        font-family: 'Open Sans', 'Clear Sans', 'Helvetica Neue', Helvetica, Arial, 'Segoe UI Emoji', sans-serif;
        font-size: 16px;
        font-style: normal;
        font-variant-caps: normal;
        font-weight: 400;
        letter-spacing: normal;
        orphans: 4;
        text-align: start;
        text-transform: none;
        white-space: pre-wrap;
        widows: auto;
        word-spacing: 0px;
        -webkit-text-size-adjust: auto;
        -webkit-text-stroke-width: 0px;
        text-decoration: none;
    "
>
</span>
<span
    md-inline="link"
    class="md-meta-i-c md-link"
    style="
        box-sizing: border-box;
        text-indent: 0px;
        caret-color: rgb(51, 51, 51);
        color: rgb(51, 51, 51);
        font-family: 'Open Sans', 'Clear Sans', 'Helvetica Neue', Helvetica, Arial, 'Segoe UI Emoji', sans-serif;
        font-size: 16px;
        font-style: normal;
        font-variant-caps: normal;
        font-weight: 400;
        letter-spacing: normal;
        orphans: 4;
        text-align: start;
        text-transform: none;
        white-space: pre-wrap;
        widows: auto;
        word-spacing: 0px;
        -webkit-text-size-adjust: auto;
        -webkit-text-stroke-width: 0px;
        text-decoration: none;
    "
>
    <a href="td-link" style="box-sizing: border-box; cursor: pointer; color: rgb(65, 131, 196); -webkit-user-drag: none; text-indent: 0px;">
        <span md-inline="plain" class="md-plain" style="box-sizing: border-box; text-indent: 0px;">td</span>
    </a>
</span>
<span
    md-inline="plain"
    class="md-plain"
    style="
        box-sizing: border-box;
        text-indent: 0px;
        caret-color: rgb(51, 51, 51);
        color: rgb(51, 51, 51);
        font-family: 'Open Sans', 'Clear Sans', 'Helvetica Neue', Helvetica, Arial, 'Segoe UI Emoji', sans-serif;
        font-size: 16px;
        font-style: normal;
        font-variant-caps: normal;
        font-weight: 400;
        letter-spacing: normal;
        orphans: 4;
        text-align: start;
        text-transform: none;
        white-space: pre-wrap;
        widows: auto;
        word-spacing: 0px;
        -webkit-text-size-adjust: auto;
        -webkit-text-stroke-width: 0px;
        text-decoration: none;
    "
>
</span>
<span
    md-inline="link"
    class="md-meta-i-c md-link"
    style="
        box-sizing: border-box;
        text-indent: 0px;
        caret-color: rgb(51, 51, 51);
        color: rgb(51, 51, 51);
        font-family: 'Open Sans', 'Clear Sans', 'Helvetica Neue', Helvetica, Arial, 'Segoe UI Emoji', sans-serif;
        font-size: 16px;
        font-style: normal;
        font-variant-caps: normal;
        font-weight: 400;
        letter-spacing: normal;
        orphans: 4;
        text-align: start;
        text-transform: none;
        white-space: pre-wrap;
        widows: auto;
        word-spacing: 0px;
        -webkit-text-size-adjust: auto;
        -webkit-text-stroke-width: 0px;
        text-decoration: none;
    "
>
    <a href="tr-link" style="box-sizing: border-box; cursor: pointer; color: rgb(65, 131, 196); -webkit-user-drag: none; text-indent: 0px;">
        <span md-inline="plain" class="md-plain" style="box-sizing: border-box; text-indent: 0px;">tr</span>
    </a>
</span>
<span
    md-inline="plain"
    class="md-plain"
    style="
        box-sizing: border-box;
        text-indent: 0px;
        caret-color: rgb(51, 51, 51);
        color: rgb(51, 51, 51);
        font-family: 'Open Sans', 'Clear Sans', 'Helvetica Neue', Helvetica, Arial, 'Segoe UI Emoji', sans-serif;
        font-size: 16px;
        font-style: normal;
        font-variant-caps: normal;
        font-weight: 400;
        letter-spacing: normal;
        orphans: 4;
        text-align: start;
        text-transform: none;
        white-space: pre-wrap;
        widows: auto;
        word-spacing: 0px;
        -webkit-text-size-adjust: auto;
        -webkit-text-stroke-width: 0px;
        text-decoration: none;
    "
>
</span>
<span
    md-inline="link"
    class="md-meta-i-c md-link"
    style="
        box-sizing: border-box;
        text-indent: 0px;
        caret-color: rgb(51, 51, 51);
        color: rgb(51, 51, 51);
        font-family: 'Open Sans', 'Clear Sans', 'Helvetica Neue', Helvetica, Arial, 'Segoe UI Emoji', sans-serif;
        font-size: 16px;
        font-style: normal;
        font-variant-caps: normal;
        font-weight: 400;
        letter-spacing: normal;
        orphans: 4;
        text-align: start;
        text-transform: none;
        white-space: pre-wrap;
        widows: auto;
        word-spacing: 0px;
        -webkit-text-size-adjust: auto;
        -webkit-text-stroke-width: 0px;
        text-decoration: none;
    "
>
    <a href="th-link2" style="box-sizing: border-box; cursor: pointer; color: rgb(65, 131, 196); -webkit-user-drag: none; text-indent: 0px;">
        <span md-inline="plain" class="md-plain" style="box-sizing: border-box; text-indent: 0px;">th_more_stuff</span>
    </a>
</span>
<span
    md-inline="plain"
    class="md-plain"
    style="
        box-sizing: border-box;
        text-indent: 0px;
        caret-color: rgb(51, 51, 51);
        color: rgb(51, 51, 51);
        font-family: 'Open Sans', 'Clear Sans', 'Helvetica Neue', Helvetica, Arial, 'Segoe UI Emoji', sans-serif;
        font-size: 16px;
        font-style: normal;
        font-variant-caps: normal;
        font-weight: 400;
        letter-spacing: normal;
        orphans: 4;
        text-align: start;
        text-transform: none;
        white-space: pre-wrap;
        widows: auto;
        word-spacing: 0px;
        -webkit-text-size-adjust: auto;
        -webkit-text-stroke-width: 0px;
        text-decoration: none;
    "
>
</span>
<span
    md-inline="link"
    class="md-meta-i-c md-link md-expand"
    style="
        box-sizing: border-box;
        text-indent: 0px;
        caret-color: rgb(51, 51, 51);
        color: rgb(51, 51, 51);
        font-family: 'Open Sans', 'Clear Sans', 'Helvetica Neue', Helvetica, Arial, 'Segoe UI Emoji', sans-serif;
        font-size: 16px;
        font-style: normal;
        font-variant-caps: normal;
        font-weight: 400;
        letter-spacing: normal;
        orphans: 4;
        text-align: start;
        text-transform: none;
        white-space: pre-wrap;
        widows: auto;
        word-spacing: 0px;
        -webkit-text-size-adjust: auto;
        -webkit-text-stroke-width: 0px;
        text-decoration: none;
    "
>
    <a href="td-link2" style="box-sizing: border-box; cursor: pointer; color: rgb(65, 131, 196); -webkit-user-drag: none; text-indent: 0px;">
        <span md-inline="plain" class="md-plain" style="box-sizing: border-box; text-indent: 0px;">td_more_stuff</span>
    </a>
</span>

```

**Comment by abnerlee** [comment]
I can reproduce this in itch.io, but not able be reproduce in other HTML editors, including https://imperavi.com/redactorx/ (itch.io seems using redactor editor)

So I think there should be something wrong for redactor editor when parsing valid HTML contents, not on Typora side.

Plus, you can "copy as HTML code" in Typora and paste in redactor's source code mode as current work around.
